src/ui/public/filter_bar/query_filter.js

```diff
@@ -83,6 +84,31 @@ define(function (require) {
     };
 
     /**
+    * Updates an existing filter
+    * @param {object} Contains a reference to a filter and its new model
+    * @returns {object} Promise that resolves to the new filter on a successful merge
+    */
+    queryFilter.updateFilter = function ({ source, model, type}) {
+      var editedFilter;
+      try {
+        editedFilter = JSON.parse(model);
```

Comment:
We should continuously validate the form and disable the **Done** button whenever the JSON isn't valid, then we shouldn't need the try/catch.